pkg/operator/ceph/object/objectstore.go

```diff
@@ -625,20 +625,28 @@ func deletePools(context *Context, spec cephv1.ObjectStoreSpec, lastStore bool)
 	return nil
 }
 
+func allObjectPools() []string {
+	pools := append(metadataPools, dataPoolName, rootPool)
+	return pools
+}
+
+func missingPools(context *Context) []string {
+	missingPools := []string{}
+	for _, pool := range allObjectPools() {
+		poolName := poolName(context.Name, pool)
+		_, err := cephclient.GetPoolDetails(context.Context, context.clusterInfo, poolName)
+		if err != nil {
+			logger.Debugf("failed to find pool %q. %v", poolName, err)
```

Comment:
If the mons are out of quorum (or some other random issue), the pools will be considered missing? Seems like this could lead to allowing uninstall even though the cluster is just not responding?

Reply:
Hmm... that's a good point. Another thought I had was to add an additional condition to the status like `DeletionProceeding = True` once deletion began. If that condition is present, it means that we can shortcut checking for dependencies to resume deletion.

I don't think we will have this same race condition for other resources, but we should be aware of the possibility for `CephFilesystems` and `CephBlockPools`.